Jucer2Reprojucer/main.cpp

```diff
@@ -270,6 +270,51 @@ void writeUserNotes(LineWriter& wLn, const juce::ValueTree& valueTree)
   }
 }
 
+    juce::ValueTree sanitizeConfigurationNames(const juce::String& exporterName, const juce::ValueTree& exporterSettingsWithInvalidConfigNames)
+    {
+        // Removes invalid characters from configuration names and adds digits to the names until all
+        // configuration names within an exporter are unique.
+        // Valid config names match the regex ^[A-Za-z0-9_]*$
+
+        const char* validCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
+
+        auto exporter = exporterSettingsWithInvalidConfigNames.createCopy();
+        juce::StringArray sanitizedConfigNames;
+
+        for (int i = 0; i < exporter.getNumChildren(); i++)
+        {
+            // remove invalid characters
+            auto configuration = exporter.getChild(i);
+            auto originalName = configuration.getProperty("name").toString();
+            auto sanitizedName = originalName.retainCharacters(validCharacters);
```

Comment:
Removing invalid characters does the job, but it's not very nice for configuration names like `Debug (all warnings)` or `Release (with LTO)`. 

Do you think you could come up with an algorithm that:
- replace invalid characters with `_`,
- keep at most one consecutive replacement `_` (i.e. it's fine to have several consecutive `_` characters if they were already in the original configuration name),
- don't add a replacement `_` at the end?

That would make `Debug (all warnings)` -> `Debug_all_warnings` and `Release (with LTO)` -> `Release_with_LTO`

Reply:
Sure, good idea

Reply:
`makeValidIdentifier` seems to do most of this already, except than only some characters are replaced with `_`, other are simply removed (this includes the space).

Now that we're dealing with multiple name-sanitization functions, it would be good to a) merge them into one or b) alter their names to be more clear about their use.
Forthe sake of readability and simplicity I would prefer to use `makeValidIdentifier` for the config name sanitization as well and add the required changes to this function instead of writing another one.

Do you agree?

Reply:
Jucer2Reprojucer's `makeValidIdentifier` is a stripped-down version of [a function named the same in Projucer](https://github.com/WeAreROLI/JUCE/blob/5.4.4/extras/Projucer/Source/Utility/Helpers/jucer_CodeHelpers.cpp#L71-L123). It is then used in Jucer2Reprojucer exactly like in Projucer. We shouldn't change the logic of this function, otherwise Jucer2Reprojucer will diverge from Projucer.  I agree that there should be some documentation (or at least a comment) on the function to explain that.

We could indeed use `makeValidIdentifier` to sanitize configuration names, but it wouldn't be as nice as the algorithm I proposed (`Debug (all warnings)` would become `Debugallwarnings`). 

If you are concerned about code reuse, don't worry, this new function to convert a string into a (nice looking) valid CMake identifier will be perfect to compute `escapedJucerFileName` as well:
https://github.com/McMartin/FRUT/blob/dfd36f97e0c654f379d3f7a6034d50be2efef326/Jucer2Reprojucer/main.cpp#L546-L552